### PR TITLE
`WarmableInstance` abstraction

### PR DESCRIPTION
**Pull request by fkorotkov**
Related to #697. I think this will be simplest implementation of the functionality we want.

### CODE REVIEW

File: internal/executor/instance/persistentworker/isolation/tart/tart.go [hunk 1]
```diff
@@ -88,53 +93,26 @@ func (tart *Tart) Run(ctx context.Context, config *runconfig.RunConfig) (err err
 		})
 	}
 
-	tmpVMName := fmt.Sprintf("%s%d-", vmNamePrefix, config.TaskID) + uuid.NewString()
+	tmpVMName := fmt.Sprintf("%s-warm-", vmNamePrefix) + uuid.NewString()
```
Comment (edigaryev) [inline internal/executor/instance/persistentworker/isolation/tart/tart.go:96]: This seems to add the `-warm-` string to *all* Tart instances and not just the warm ones.

Should we add an additional argument to `Warmup()` that will specify this optional string to be added to VM names in https://github.com/cirruslabs/cirrus-cli/pull/700?

Comment (fkorotkov) [reply internal/executor/instance/persistentworker/isolation/tart/tart.go:96]: Good idea. Maybe allow to optinally specify `vmNamePrefix` as `"cirrus-warm"` vs the default `"cirrus"`?

Comment (edigaryev) [reply internal/executor/instance/persistentworker/isolation/tart/tart.go:96]: Added in https://github.com/cirruslabs/cirrus-cli/pull/700/commits/b5590f927f7a17ba0c6fc2586a82aa04a45e80eb.

File: internal/executor/instance/persistentworker/isolation/tart/tart.go [hunk 2]
```diff
@@ -192,22 +188,47 @@ func (tart *Tart) Run(ctx context.Context, config *runconfig.RunConfig) (err err
...
 
 	bootLogger.Errorf("VM was assigned with %s IP", ip)
 	bootLogger.Finish(true)
 
+	return nil
+}
+
+func (tart *Tart) Run(ctx context.Context, config *runconfig.RunConfig) (err error) {
+	if tart.vm != nil && config.DirtyMode {
+		return fmt.Errorf("%w: dirty mode is not supported for a warmed instance", ErrFailed)
```
Comment (edigaryev) [inline internal/executor/instance/persistentworker/isolation/tart/tart.go:203]: Not all Tart instances will be standby instances (this assumption was also in https://github.com/cirruslabs/cirrus-cli/pull/698#discussion_r1523350920).

I think we still have a `cirrus run --dirty` (https://github.com/cirruslabs/cirrus-cli/issues/554) or are we dropping it for Tart?

File: internal/executor/instance/persistentworker/isolation/tart/tart.go [hunk 3]
```diff
@@ -192,22 +188,47 @@ func (tart *Tart) Run(ctx context.Context, config *runconfig.RunConfig) (err err
 		break
 	}
 
+	tart.vm = vm
+
 	tart.logger.Debugf("IP %s retrieved from VM %s, running agent...", ip, vm.Ident())
```
Comment (edigaryev) [inline internal/executor/instance/persistentworker/isolation/tart/tart.go:193]: I think this should be in `Run()`, we're not running the agent here.

File: internal/executor/instance/persistentworker/isolation/tart/tart.go (hunk 2 above)
Comment (fkorotkov) [reply internal/executor/instance/persistentworker/isolation/tart/tart.go:203]: There is no dirty mode for persistent workers so this should never happen. Warmable instances won't be used for `cirrus run`

File: internal/executor/instance/persistentworker/isolation/tart/tart.go (hunk 3 above)
Comment (fkorotkov) [reply internal/executor/instance/persistentworker/isolation/tart/tart.go]: Moved in [04fee80](https://github.com/cirruslabs/cirrus-cli/pull/698/commits/04fee80eef05acaff472362efe61466cd1985116)

**Review by edigaryev — APPROVED** [review]

**Pull request merged**